src/LibraryManager.Vsix/Json/TextviewCreationListener.cs

```diff
@@ -107,22 +107,27 @@ public void VsTextViewCreated(IVsTextView textViewAdapter)
 
         private void OnFileSaved(object sender, TextDocumentFileActionEventArgs e)
         {
-            if (e.FileActionType == FileActionTypes.ContentSavedToDisk)
+            var textDocument = (ITextDocument) sender;
+
+            if (e.FileActionType == FileActionTypes.ContentSavedToDisk && textDocument != null)
             {
                 Task.Run(async () =>
                 {
                     try
                     {
-                        Manifest newManifest = await Manifest.FromFileAsync(e.FilePath, _dependencies, CancellationToken.None).ConfigureAwait(false);
+                        Manifest newManifest = Manifest.FromJson(textDocument.TextBuffer.CurrentSnapshot.GetText(), _dependencies);
                         await RemoveFilesAsync(newManifest).ConfigureAwait(false);
 
                         _manifest = newManifest;
 
-                        await LibraryHelpers.RestoreAsync(e.FilePath, CancellationToken.None).ConfigureAwait(false);
+                        await LibraryHelpers.RestoreAsync(textDocument.FilePath, _manifest, CancellationToken.None).ConfigureAwait(false);
                         Telemetry.TrackOperation("restoresave");
                     }
                     catch (Exception ex)
                     {
+                        string textMessage = string.Concat(Environment.NewLine, LibraryManager.Resources.Text.RestoreHasErrors, Environment.NewLine);
```

Comment:
Adding logging to Output pane and status bar in case of failure. This is the same behavior we have for other restore operations invoked by UI.